core/server/app_test.go

```diff
@@ -0,0 +1,133 @@
+package server
```

Comment:
this should probably be `server_test`, but I can see that the rest of the files in this suite are written as internal tests although the interfaces are external facing, so this could lead to the contract boundary being blurred. do you know the history of this? it may be irrelevant: the clearance team will probs get to it.

Reply:
We have been using internal tests as a convention. My understanding is that this is the current flavor for the `go` community. Being able to write tests for complex un-exported funcs seems like an advantage; any strong feelings either way?

Reply:
I tend to lean towards writing exported tests wherever possible as it encourages me to design along clean lines, keep things in isolated units, maintain clear "contracts" between components, etc. I like internal tests for utils or helper packages, basically things which have no interface or are not designed to be consumed by anything else.
I think in this case since all the funcs being tested are exported, and it is only the `appServer` object which isn't, we could probably test the package externally.

But I am new to this code so I am fine going with and coming to understand convention and don't need to double down on any opinions right now.